### ISSUE TITLE
When calling force refresh, failure to write on cache will make it impossible to refresh again without server state change

### ISSUE THREAD

**Issue by ls-rein-martha** [configcat/java-sdk#12]
### Describe the bug

Issue on calling `forceRefresh()` when previous refresh fail on writing to cache. With custom `ConfigCache`.
It is possible to fail when writing to `cache`, example: distributed cache can fail to write (even though the chance is very small it still exist).

### To reproduce

Use `MANUAL` polling mode.
Create a feature flag, lets create `String` for easy check, with initial value: "INITIAL"
Create a custom ConfigCache, like below to mimic exception:
```
    public class FailWriteEveryThirdRequest extends ConfigCache{
        Map<String, String> map = new HashMap<>();
        AtomicInteger counter = new AtomicInteger(1);
        
        @Override
        protected String read(String key) throws Exception {
            return map.get(key);
        }

        @Override
        protected void write(String key, String value) throws Exception {
            if(counter.getAndIncrement() % 2 == 0){
                throw new RuntimeException("I fail for some reason"); // just to mimic exception on failure writing
            }
            map.put(key, value);
        }
    }
```
Update the value and refresh -> everything good -> before refresh for the second time (that will throw exception): 
- Update value to "UPDATE WILL THROW EXCEPTION"
- Since it will throw exception cache won't be updated

Do force refresh again:
- Value will not be updated because it return `NOT_MODIFED` response and it won't write to cache
- When GET the flag it will still return the previous value from cache.
- Doing `forceRefresh()` is not doing anything now since there is no changes at server. (Not sure if it will be different after some time)

Root cause:
- I am not sure how the server check the "version" of the client feature value, but i believe it based on the "ETag"? 
- As you can see [here](https://github.com/configcat/java-sdk/blob/master/src/main/java/com/configcat/ConfigFetcher.java#L126) it save the "ETag" value.
- But at the exception handling on cache writing [here](https://github.com/configcat/java-sdk/blob/master/src/main/java/com/configcat/ConfigJsonCache.java#L67) it doesn't revert the "ETag" thus I believe server thought its already updated.

### Expected behavior

At minimum, when fail to write to cache, the next successful refresh (done manually) should be able to update the value.
Better solution is server can know when there is a problem on writing and when client use webhook, server can do retry mechanism.

### Screenshots

N/A

### SDK version

Tried at 7.0.4 and 7.0.5

### SDK configuration

Default configuration also can reproduce. (Just use manual polling and manual force refresh)

### Logs

N/A

### Language/Framework version

Java 8

### Platform

Any

**Comment by z4kn4fein** [comment]
Hi @ls-rein-martha, thank you for reporting this issue! I'm going to start working on this and I'll let you know when the fix is ready.

**Comment by z4kn4fein** [comment]
Hi @ls-rein-martha, I released the fix for this issue in v7.0.6, could you please check that it works now on your end as expected?
Thanks!

**Comment by ls-rein-martha** [comment]
> Hi @ls-rein-martha, I released the fix for this issue in v7.0.6, could you please check that it works now on your end as expected? Thanks!

Hi @z4kn4fein thank you for quick fix. I will wait for the maven repo jar to test, I have not see it available yet [there](https://mvnrepository.com/artifact/com.configcat/configcat-java-client). :)

**Comment by ls-rein-martha** [comment]
Tested. Fixed at v7.0.6.
